src/core_codemods/refactor/refactor_new_api.py

```diff
@@ -0,0 +1,297 @@
+import libcst as cst
+
+from core_codemods.api import SimpleCodemod, Metadata, ReviewGuidance
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+
+
+class RefactorNewApi(SimpleCodemod, NameResolutionMixin):
```

Comment:
this clearly took some effort so I can feel the reason to commit it, but do you think we will put it to use later on? maybe add a docstring explaining this was used during this transition.

Reply:
I can add a docstring. It is one-time use but it's also a nice demonstration of the capabilities of codemods. I intend to blog about it later. It also genuinely saved me some time with refactoring ~35 of the codemods.